RevenueCatUI/Templates/Components/Stack/StackComponentView.swift

```diff
@@ -22,28 +22,66 @@ struct StackComponentView: View {
-                ComponentsView(componentViewModels: self.viewModel.viewModels)
+        Group {
+            switch viewModel.dimension {
+            case .vertical(let horizontalAlignment):
+                VStack(spacing: viewModel.spacing) {
+                    Group {
+                        ComponentsView(componentViewModels: self.viewModel.viewModels)
+                    }
+                    .frame(maxWidth: .infinity, alignment: horizontalAlignment.stackAlignment)
+                }
+            case .horizontal(let verticalAlignment):
+                HStack(alignment: verticalAlignment.stackAlignment, spacing: viewModel.spacing) {
+                    ComponentsView(componentViewModels: self.viewModel.viewModels)
+                }
+            case .zlayer(let alignment):
+                ZStack(alignment: alignment.stackAlignment) {
+                    ComponentsView(componentViewModels: self.viewModel.viewModels)
+                }
             }
-            .background(viewModel.backgroundColor)
-            .padding(viewModel.padding)
-        case .horizontal(let verticalAlignment):
-            HStack(alignment: verticalAlignment.stackAlignment, spacing: viewModel.spacing) {
-                ComponentsView(componentViewModels: self.viewModel.viewModels)
-            }
-            .background(viewModel.backgroundColor)
-            .padding(viewModel.padding)
-        case .zlayer(let alignment):
-            ZStack(alignment: alignment.stackAlignment) {
-                ComponentsView(componentViewModels: self.viewModel.viewModels)
+        }
+        .padding(viewModel.padding)
+        .width(viewModel.width)
+        .background(viewModel.backgroundColor)
+        .roundedCorner(viewModel.cornerRadiuses.topLeading, corners: .topLeft)
+        .roundedCorner(viewModel.cornerRadiuses.topTrailing, corners: .topRight)
+        .roundedCorner(viewModel.cornerRadiuses.bottomLeading, corners: .bottomLeft)
+        .roundedCorner(viewModel.cornerRadiuses.bottomTrailing, corners: .bottomRight)
+        .padding(viewModel.margin)
+    }
+
+}
+
+struct WidthModifier: ViewModifier {
+    var width: PaywallComponent.WidthSize?
+
+    func body(content: Content) -> some View {
+        if let width = width {
+            switch width.type {
+            case .fit:
+                content
+            case .fill:
+                content
+                    .frame(maxWidth: .infinity)
+            case .fixed:
+                if let value = width.value {
+                    content
+                        .frame(width: CGFloat(value))
+                } else {
+                    content
+                }
             }
-            .background(viewModel.backgroundColor)
-            .padding(viewModel.padding)
+        } else {
+            content
         }
     }
+}
 
+extension View {
+    func width(_ width: PaywallComponent.WidthSize? = nil) -> some View {
```

Comment:
nitpick: add line above and after the function to match our Swift style guide. Same for `WidthModifier`, adding lines both at the beginning and at the end